frontend/src/employee-mobile-frontend/staff-attendance/StaffAttendanceEditPage.tsx

```diff
@@ -0,0 +1,423 @@
+                  })
+                ])
+              }}
+              data-qa="add"
+            >
+              {i18n.attendances.staff.add}
+            </ButtonLink>
+          )}
+      </ContentArea>
+      <ContentArea opaque paddingHorizontal="s">
+        <FixedSpaceRow justifyContent="space-between">
+          <Button
+            data-qa="cancel"
+            onClick={() =>
+              navigate(
+                `${groupRoute}/staff-attendance/${staffMember.employeeId}`
+              )
+            }
+          >
+            {i18n.common.cancel}
+          </Button>
+          <div>
+            <Label>{i18n.pin.pinCode}</Label>
+            <PlainPinInputF
+              bind={pinCode}
+              info={
+                errorCode !== undefined && errorCode in i18n.pin.status
+                  ? {
+                      text: i18n.pin.status[errorCode as PinLoginStatus],
+                      status: 'warning'
+                    }
+                  : undefined
+              }
+            />
+            <Gap size="s" />
+            <MutateButton
+              primary
+              data-qa="save"
+              text={i18n.common.saveChanges}
+              mutation={staffAttendanceMutation}
+              onClick={() => ({
+                unitId,
+                request: {
+                  employeeId,
+                  pinCode: pinCode.value(),
+                  date,
+                  rows: rows.value()
+                }
+              })}
+              onSuccess={() =>
+                navigate(
+                  `${groupRoute}/staff-attendance/${staffMember.employeeId}`
+                )
+              }
+              onFailure={({ errorCode }) => {
+                setErrorCode(errorCode)
+                if (errorCode === 'WRONG_PIN') {
+                  pinCode.update(() => '')
+                }
+              }}
+              disabled={!form.isValid()}
+            />
+          </div>
+        </FixedSpaceRow>
+      </ContentArea>
+    </>
+  )
+}
+
+const StaffAttendanceRowEditor = ({
+  date,
+  rows,
+  index
+}: {
+  date: LocalDate
+  groups: GroupInfo[]
+  rows: BoundForm<typeof staffAttendanceRows>
+  index: number
+}) => {
+  const { i18n, lang } = useTranslation()
+  const form = useFormElem(rows, index)
+  if (form === undefined) {
+    throw new Error('index error')
+  }
+  const { type, groupId, arrivedDate, arrivedTime, departedTime } =
+    useFormFields(form)
+
+  const isArrivedPastDate = arrivedDate.value().isBefore(date)
+
+  return (
+    <FixedSpaceRow alignItems="baseline" flexWrap="wrap">
+      {featureFlags.staffAttendanceTypes && (
+        <div>
+          <SelectF bind={type} data-qa="type" />
+          {type.validationError() && (
+            <InputFieldUnderRow className={classNames('warning')}>
+              {type.inputInfo()?.text}
+              <StatusIcon status="warning" />
+            </InputFieldUnderRow>
+          )}
```

Comment:
I wonder if it would be good to add validation error showing directly to `Select`/`SelectF`, because many other form components already can do it. It beats the idea of the form framework a bit if individual components don't display their errors and you have to remember to do it in the parent.